File: z_lc/1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        #include <unordered_map>
        
        unordered_map<int, int> seen;
        for(int i=0; i<nums.size(); ++i) {
            int n = nums[i];
            int complement = target - n;
            if(auto iter = seen.find(complement); iter != seen.end()) {
                return vector<int> {i, iter->second};    
            }
            seen[n] = i;
        }
        return vector<int>();
    }
};

File: z_lc/1-two-sum/two-sum.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        use std::collections::HashMap;
        
        let mut seen = HashMap::new();
        for (i,n) in nums.iter().enumerate() {
            let complement = target - n;
            match seen.get(&complement) {
                Some(&index) => return vec![i as i32, index],
                None => seen.insert(n, i as i32),
            };
        }
        vec![]
    }
}
